sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
+      printf("A twin reported properties response message was received\n");
+      break;
+  }
+}
+
+// Invoke the requested command if supported and return status | Return error otherwise
+static void handle_command_message(
+    MQTTClient_message* message,
+    az_iot_hub_client_method_request* command_request)
+{
+
+  if (az_span_is_content_equal(report_command_name_span, command_request->name))
+  {
+    az_span command_response_span = AZ_SPAN_FROM_BUFFER(commands_response_payload);
+    az_span command_payload_span
+        = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+
+    // Invoke command
+    uint16_t return_code;
+    az_result response = invoke_getMaxMinReport(
+        command_payload_span, command_response_span, &command_response_span);
+    if (response == AZ_ERROR_ITEM_NOT_FOUND)
+    {
+      return_code = 404;
+    }
+    else
+    {
+      return_code = 200;
+    }
+
+    // Send command response with report as JSON payload
+    int rc;
+    if ((rc = send_command_response(command_request, return_code, command_response_span)) != 0)
+    {
+      printf("Unable to send %u response, status %d\n", return_code, rc);
+    }
+  }
+  else
+  {
+    // Unsupported command
+    printf(
+        "Unsupported command received: %.*s.\n",
+        az_span_size(command_request->name),
+        az_span_ptr(command_request->name));
+
+    int rc;
+    if ((rc = send_command_response(command_request, 501, report_error_payload)) != 0)
+    {
+      printf("Unable to send %d response, status %d\n", 501, rc);
+    }
+  }
+}
+
+// Callback for incoming MQTT messages
+static int on_received(char* topicName, int topicLen, MQTTClient_message* message)
+{
+  if (topicLen == 0)
+  {
+    // The length of the topic if there are one or more NULL characters embedded in topicName,
+    // otherwise topicLen is 0.
+    topicLen = (int)strlen(topicName);
+  }
+
+  printf("Topic: %s\n", topicName);
+
+  az_span topic_span = az_span_init((uint8_t*)topicName, topicLen);
+
+  // Parse the incoming message topic and check which feature it is for
+  az_iot_hub_client_twin_response twin_response;
+  az_iot_hub_client_method_request command_request;
+
+  if (az_succeeded(
+          az_iot_hub_client_twin_parse_received_topic(&client, topic_span, &twin_response)))
+  {
+    printf("Twin Message Arrived\n");
+
+    // Determine what kind of twin message it is and take appropriate actions
+    handle_twin_message(message, &twin_response);
+
+    printf("Response status was %d\n", twin_response.status);
+  }
+  else if (az_succeeded(az_iot_hub_client_methods_parse_received_topic(
+               &client, az_span_init((uint8_t*)topicName, topicLen), &command_request)))
+  {
+    printf("Command arrived\n");
+
+    // Determine if the command is supported and take appropriate actions
+    handle_command_message(message, &command_request);
+  }
+
+  putchar('\n');
```

Comment:
>putchar('\n'); [](start = 2, length = 14)

Nit: do we really need this putchar() here as printf()'s end in "\n" from what i see?

Reply:
It separates it on the console into more logical blocks instead of being one long continuous block of text. So readability